File: src/main/java/com/kamegatze/learnjvm/repository/roles/RolesRepository.java
package com.kamegatze.learnjvm.repository.roles;

import com.kamegatze.learnjvm.model.db.roles.ERoles;
import com.kamegatze.learnjvm.model.db.roles.Roles;
import org.jetbrains.annotations.NotNull;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface RolesRepository extends JpaRepository<Roles, UUID> {

    Optional<Roles> findByName(ERoles name);

    @Override
    default void deleteAllInBatch() {
        throw new UnsupportedOperationException("Not supported yet.");
    }

    @Override
    default void deleteAllByIdInBatch(@NotNull Iterable<UUID> uuids) {
        throw new UnsupportedOperationException("Not supported yet.");
    }

    @Override
    default void deleteAllInBatch(@NotNull Iterable<Roles> entities) {
        throw new UnsupportedOperationException("Not supported yet.");
    }

    @Override
    default <S extends Roles> @NotNull List<S> saveAllAndFlush(@NotNull Iterable<S> entities) {
        throw new UnsupportedOperationException("Not supported yet.");
    }

    @Override
    default <S extends Roles> @NotNull S saveAndFlush(@NotNull S entity) {
        throw new UnsupportedOperationException("Not supported yet.");
    }

    @Override
    default void flush() {
        throw new UnsupportedOperationException("Not supported yet.");
    }

    @Override
    default <S extends Roles> @NotNull List<S> saveAll(@NotNull Iterable<S> entities) {
        throw new UnsupportedOperationException("Not supported yet.");
    }

    @Override
    default void deleteInBatch(@NotNull Iterable<Roles> entities) {
        throw new UnsupportedOperationException("Not supported yet.");
    }

    @Override
    default <S extends Roles> @NotNull S save(@NotNull S entity) {
        throw new UnsupportedOperationException("Not supported yet.");
    }

    @Override
    default void deleteById(@NotNull UUID uuid) {
        throw new UnsupportedOperationException("Not supported yet.");
    }

    @Override
    default void delete(@NotNull Roles entity) {
        throw new UnsupportedOperationException("Not supported yet.");
    }

    @Override
    default void deleteAllById(@NotNull Iterable<? extends UUID> uuids) {
        throw new UnsupportedOperationException("Not supported yet.");
    }

    @Override
    default void deleteAll(@NotNull Iterable<? extends Roles> entities) {
        throw new UnsupportedOperationException("Not supported yet.");
    }

    @Override
    default void deleteAll() {
        throw new UnsupportedOperationException("Not supported yet.");
    }
}

File: src/main/kotlin/com/kamegatze/learnjvm/repositories/roles/RolesRepository.kt
package com.kamegatze.learnjvm.repositories.roles

import com.kamegatze.learnjvm.model.db.roles.Roles
import org.springframework.data.repository.CrudRepository
import org.springframework.data.repository.PagingAndSortingRepository
import java.util.*

interface RolesRepository : CrudRepository<Roles, UUID>, PagingAndSortingRepository<Roles, UUID> {
    fun findByName(name: String): Roles?

    override fun <S : Roles?> save(entity: S & Any): S & Any {
        throw UnsupportedOperationException()
    }

    override fun <S : Roles?> saveAll(entities: MutableIterable<S>): MutableIterable<S> {
        throw UnsupportedOperationException()
    }

    override fun deleteById(id: UUID) {
        throw UnsupportedOperationException()
    }

    override fun delete(entity: Roles) {
        throw UnsupportedOperationException()
    }

    override fun deleteAllById(ids: MutableIterable<UUID>) {
        throw UnsupportedOperationException()
    }

    override fun deleteAll(entities: MutableIterable<Roles>) {
        throw UnsupportedOperationException()
    }

    override fun deleteAll() {
        throw UnsupportedOperationException()
    }
}